scratch_4.kts

```diff
@@ -0,0 +1,37 @@
+abstract class car{
```

Comment:
Trong trường hợp này, e nghĩ sử dụng `abstract class`, thay vì interface hoặc class thường có ưu nhược điểm gì?

Reply:
em thấy abstract class vừa có thể để biến và để phương thức còn interface chỉ để được phương thức
còn về kiểu open class thì em thấy nó gần như bao hàm của cả 2 cái trên

Reply:
uh, nhìn chung thì e trả lời tương đối đúng. Có điều interface ngoài phương thức thì mình cũng để được cả thuộc tính nữa. Khác biệt lớn nhất giữa 3 thằng này là có cái e có thể implement thân hàm, hoặc/và gán giá trị cho thuộc tính, còn có cái thì  không được, và class kế thừa/ implement nó phải triền khai phương thức/thuộc tính đó

Reply:
cũng vì nó khác nhau như vậy, nên tùy từng trường hợp, mình mới sử dụng từng cái. Có bài toán sử dụng open class thì hợp lý, nhưng ở chỗ khác thì abstract class lại là lựa chọn tốt hơn

Reply:
ở đây `abstract class Car` của e ko mô tả sẵn thuộc tính & phương thức, nên sử dụng 1 abstract class chưa đc hợp lý cho lắm